test/infrastructure/docker/api/v1beta1/conversion.go

```diff
@@ -128,3 +129,245 @@ func Convert_v1beta1_ObjectMeta_To_v1beta2_ObjectMeta(in *clusterv1beta1.ObjectM
 func Convert_v1beta2_ObjectMeta_To_v1beta1_ObjectMeta(in *clusterv1.ObjectMeta, out *clusterv1beta1.ObjectMeta, s apiconversion.Scope) error {
 	return clusterv1beta1.Convert_v1beta2_ObjectMeta_To_v1beta1_ObjectMeta(in, out, s)
 }
+
+func Convert_v1beta1_DevClusterStatus_To_v1beta2_DevClusterStatus(in *DevClusterStatus, out *infrav1.DevClusterStatus, s apiconversion.Scope) error {
+	if err := autoConvert_v1beta1_DevClusterStatus_To_v1beta2_DevClusterStatus(in, out, s); err != nil {
+		return err
+	}
+
+	// Reset conditions from autogenerated conversions
+	// NOTE: v1beta1 conditions should not be automatically be converted into v1beta2 conditions.
+	out.Conditions = nil
+
+	// Retrieve new conditions (v1beta2) from the v1beta2 field.
+	if in.V1Beta2 != nil {
+		out.Conditions = in.V1Beta2.Conditions
+	}
+
+	// Move legacy conditions (v1beta1) to the deprecated field.
+	if in.Conditions == nil {
+		return nil
+	}
+
+	if out.Deprecated == nil {
+		out.Deprecated = &infrav1.DevClusterDeprecatedStatus{}
+	}
+	if out.Deprecated.V1Beta1 == nil {
+		out.Deprecated.V1Beta1 = &infrav1.DevClusterV1Beta1DeprecatedStatus{}
+	}
+	if in.Conditions != nil {
+		out.Deprecated.V1Beta1.Conditions = in.Conditions
+	}
```

Comment:
```suggestion
	out.Deprecated.V1Beta1.Conditions = in.Conditions
```
We already know conditions is not nil due to the test in L148

Reply:
Same for all the other Convert_v1beta1_*Status_To_v1beta2_*Status func